feat: 소관위원회 repository 추가

gzcxadfzc opened the pull request:
최대한 패키지 구조 따라가려고 햇음
이해한 바로는
Domain 외부의존성 금지
Db core에서 repository를 adapter를 통해 주입(DIP)방식

현재 코드구조는 LegislationAccount를 상속받는 LegislationCommitteeAccount(소관위계정)가 있음

궁금한것은
1. domain-controller간 dto를 사용하는지? 사용한다면 dto관련 패키징은 어떻게 할것인지?
2. controller 관련
- controller 모듈에서 응답형식 양식이 있는지 혹은 ResponseEntity그대로 사용하는지
- 그외 잡다한 패키징 명세가 있다면 알려주세오

이해한대로 일단 조금 만들어봤는데 올바르게 이해했는지 확인좀 부탁합니다(-_-)(_ _)

chobuisness committed:
add : value 에 대한 getter 추가

chobuisness committed:
feature : 법안 발의자 table mapping

chobuisness committed:
add : 법안명, 입법기관 필드 추가

chobuisness committed:
feature : 법안 게시글에 대한 필터링 태그 관리 객체

chobuisness committed:
feature : 쿼리 정렬 관리 객체

chobuisness committed:
feature : 최근 법안 게시글 repository

chobuisness committed:
feature : 최근 법안 게시글 domain entity

chobuisness committed:
feature : 최근 법안 게시글 조회 컴포넌트 및 repository

chobuisness committed:
feature : 최근 법안 게시글 paging 조회 결과

chobuisness committed:
feature : 법안 게시글 조회 쿼리 커맨드 객체

chobuisness committed:
feature : 최근 법안 게시글 조회 service 컴포넌트

chobuisness committed:
build : storage:db-core 모듈 runtime-only 의존성 추가 + core-domain 모듈 impl 의존성 추가

chobuisness committed:
config : notification.yml + db-core.yml 추가

chobuisness committed:
feature : 최근 법안 게시글 thumbnail 조회 controller

chobuisness committed:
feature : 최근 법안 게시글 response

chobuisness committed:
feature : 최근 법안 게시글 thumbnail 조회 api spec 조합기 객체

chobuisness committed:
add : profile image 필드 추가

chobuisness committed:
add : 최근 발의법안 게시글의 bill id 반환 추가

gzcxadfzc committed:
feat: 소관위원회 repository 추가

chobuisness committed:
feature : core-domain 모듈 exception

chobuisness committed:
feature : 최근 법안 게시글 상세 조회 query 커멘드

chobuisness committed:
feature : 최근 법안 게시글 도메인 exception

chobuisness committed:
feature : 법안발의자 객체 - RecentBillPost 하위 도메인 객체

chobuisness committed:
feature : 법안발의자 객체 추가

chobuisness committed:
add : 최근 법안 게시글 상세 조회 기능 추가

chobuisness committed:
feature : api response 관련 constant 객체

chobuisness committed:
feature : 최근 법안 게시글 response 및 response 생성기

chobuisness committed:
add : 최근 법안 게시글 상세 조회 추가

chobuisness committed:
add : 법안발의자 jpa entity 필드 추가

chobuisness committed:
add : PartyName enum 필드 추가

chobuisness committed:
feature : 법안발의자 jpa repository

chobuisness committed:
add : 최근 법안 게시글 상세 조회 repository adapter 추가

chobuisness committed:
add : 최근 법안 게시글 상세 조회 response 필드 추가

chobuisness committed:
feature : core-domain 위상의 예외 전파 controller advice

chobuisness reviewed (commented):
일단 의존성 방향 부분은 잘 이해한거같고, 몇 가지 커멘트 달아놨음.
그 노션에 `진행표>컨벤션` << 이거 써놨으니깐 한 번 읽어보셈 ㅋㅋ

그리고 지금 이 페이지 조회 api 개발하는걸로 알고있는데, `LegislationCommitteeAccount` 까지 필요한지는 살짝 의문임
<img width="245" alt="스크린샷 2025-01-07 오후 7 41 26" src="https://github.com/user-attachments/assets/a73d9241-c26d-4f1f-a33e-117184f3789c" />

gzcxadfzc committed:
refactor: 불필요한 클래스 및 리팩터

gzcxadfzc committed:
feat: 구독정보 repository 추가

gzcxadfzc committed:
feat: 알림정보 repository 추가

gzcxadfzc committed:
feat: 소관위원회 알림 구독 정보 조회 기능 추가

gzcxadfzc committed:
feat: 소관위원회 알림 구독 정보 조회 controller추가

auth모듈 이후 추가작업 필요

gzcxadfzc committed:
feat: 소관위원회 알림 구독 정보 조회 controller추가

auth모듈 이후 추가작업 필요

gzcxadfzc committed:
Merge branch 'legislation/feature/issue/16' of https://github.com/ogongchill/barlow into legislation/feature/issue/16

gzcxadfzc committed:
refactor: CommitteeNotificationSubscriptionComposer 변경

gzcxadfzc committed:
build: application및 빌드 설정

+ db-core.yml
+ notification.yml

gzcxadfzc committed:
test: adapter클래스 테스트 추가

+ 테스트에 따른 자잘한 버그 수정

gzcxadfzc committed:
Merge branch 'legislation/feature/issue/15' into 'legislation/feature/16'

- merge conflicts in application.yml, LegislationType
- fix BillProposerJpaRepository: change BillProposer -> BillProposerJpaEntity

gzcxadfzc committed:
refactor: 예외처리 방식 변경

- CoreDomainException 상속받아 처리

gzcxadfzc committed:
test: 소관위 구독 알림 정보 조회 관련 테스트 코드 추가

- 테스트 코드에 따른 자잘한 수정

chobuisness reviewed (commented):
일단 고생추 하나 드림 👍🏻
전체적으로 잘했는데, 몇 가지 수정하면 좋을거같은 부분이 있음

일단 지금은 #16 이슈를 처리하고 있는데, 관련 화면은 대충 아래 형식일것임.

<img width="259" alt="스크린샷 2025-01-22 오후 5 53 40" src="https://github.com/user-attachments/assets/50b0e07e-31d9-4562-b7db-0e3560b1ed47" />

- 일단 핵심 도메인이 뭔지 생각해 봐야함
- 아마 **`소관위`** 가 메인 도메인이 될 것이고, 이와 관련된 부가적인 기능들(소관위 별 페이지, 소관위 별 상세 게시글 등)이 추가될 걸로 생각됨
- 그러면 결국 우리가 집중해야할 부분은 `committee` 라는 도메인임

### 1. 패키지 구분
일단 패키지는 응집을 보여주기 위한 것임.
import 문을 보면 해당 객체가 의존하는 다른 도메인을 확인할 수 있음. 그런 의미에서 객체를 파악하기 위한 좋은 정보임. 
그런데 의존성이 있는 도메인이 필수적인지를 생각해봐야함. 
예를 들어, `CommitteeNotificationSubscriptionRetrieveService` 객체를 보면 `account`, `notification`, `subscription` 에 대한 import 가 들어와있는 것을 볼 수 있음. 그럼 이런 질문을 해볼 수 있음.
    - 얘네가 진짜 하나의 도메인으로 분리될 수 있는 놈들인가? 
    - 사용자의 구독 정보, 알림 정보를 조회하는 기능이 개별적인 사용자 요청에 의해서 수행될 수 있는 애들인가?
    - 그냥 `committee` 라는 도메인 하위에 속한 도메인이진 않은가?

이런 부분 관련해서 한번 고민해보면 좋을듯. 내 생각에는 일단 `account` 나 `committee` 중 하나의 상위 도메인만 있어도 될 것 같음. 

### 2. UI 의존 개발
일단 지금 단계에서 우리 서비스는 거의 조회(GET요청)가 대부분임. 
조회는 당연히 비즈니스 로직에 포함되지만, **조회된 결과를 화면에 어떻게 보여주는가**는 다른 문제임. 
예를 들어, `CommitteeNotificationSubscriptionComposer` 는 `CommitteeNotificationSubscription` 객체를 구성하기 위한 역할을 해주는 것 같음. 근데 `CommitteeNotificationSubscription` 이 객체가 과연 우리 핵심 도메인 객체(애그리거트)인가 생각해보면 그렇지 않음. 그냥 화면에 보여줘야하는 결과(api spec)에 맞는 형식을 객체 형식으로 만들어준 것 뿐임. 
즉, core-api 모듈에 있는`CommitteeSubscriptionResponse.AccountInfo` 와 형식이 완전히 일치하게 됨. 만약에 화면에 보여줘야하는 api-spec 형식이 달라지게 된다면? 
`CommitteeNotificationSubscription` 얘도 바꿔줘야할거고, 얘를 구성해주는 로직도 변경해줘야함. 
다시 말해서 `프리젠테이션 레이어`에서의 변경이 `도메인 레이어`의 변경으로 전파되는것. UI 변경 때문에 우리 도메인 로직이 변경되는 건 별로라고 생각함.

### 결론
결론적으로 하고싶은 말은,
- 이 이슈에서 핵심 도메인이 뭔지 한 번 생각해보고
- UI 에 의존된 개발을 하지 않도록 분리하는 방법 << `recentbill` 도메인 참고
을 생각해보면 좋을듯

gzcxadfzc committed:
refactor:  알림 조회 실패 시 예외처리 제거 및 알림,구독 조합 방식 변경

- 알림정보가 조회되지 않을시 예외처리 하지 않는 방식으로 변경
- CommitteeSubscriptionNotificationApiSpecComposer를 통해 core-api에서 조합하는 방식으로 변경

chobuisness reviewed (commented):
`ApiSpecComposer` 사용해서 앞서 말했던 UI 의존성 분리하려고 한거 굿임 ㅇㅇ
근데 본인 의견으로는 여전히 UI 의존성이 퍼져있는 부분들이 있다고 보여짐. 아래 코멘트 보면 느낌 올거 ㅋㅋ
사실 이거에 정답은 없는거라서, 어케 생각하는지 들어보고, 뭐가 더 관리가 쉬울지 or 뭐가 더 가독성이 뛰어날지 등등을 결정해서 한 가지 방법으로 통일하는게 전체 프로젝트 규칙(통일성)을 위해서도 좋을듯 ㅇㅇ

일단 본인은 기본적으로 `조회` 는 뭐 특별한 비즈니스를 담고있지 않다고 생각함.
결국 db 에서 관련 데이터 가져와서, 그걸 어떻게 화면에 뿌려주느냐가 핵심이라고 봄.
즉, 조회에서의 핵심은 **화면구성** 임.

그래서 내가 선택한 방식은 간단함.
1. 그냥 필요한 데이터를 db 에서 일단 select 해옴. 
2. domain 에서 해당 db entitiy 를 domain entity 로 감싼 뒤
3. controller 로 해당 도메인 엔티티 뿌려줌
4. controller 에서 화면 구성에 맞게 구성함
이게 내가 생각했을때, 우리가 선택한 레이어 구성에 맞는 가장 간단한 조회 로직임.
사실 물론 조회가 상당히 복잡해지고 하면,, 나도 어떻게 될지 모르겠음. 경험해본 적이 없어서 ㅋㅋ
근데 일단 초기에는 이런 방식이 제일 간단하면서 레이어 규칙에도 잘 맞는 방식이 아닐까 생각함.

그래서 지금 이 `소관위 더 알아보기` 도 제일 간단하게 하면
1. 소관위 싹 다 가져옴
2. 회원 구독 정보 가져옴
3. 알림 정보 가져옴
4. controller 에서 얘네 화면에 맞게 구성함
정도로 할 수 있지 않을까 싶음.

한 번 생각해보고 내일 얘기 ㄱㄱ

gzcxadfzc committed:
refactor: ui관련 로직 apiSpecComposer로 변경

gzcxadfzc committed:
refactor: 불필요한 검증 로직 제거

gzcxadfzc committed:
refactor: 불필요한 검증 로직 제거

chobuisness committed:
Merge branch 'base/legislation' into legislation/feature/issue/16
